Develop

Petar-Saric opened the pull request:
Тупко #29: Windows callback

Petar-Saric committed:
rearranging and adding some debugging features concerning socket error conditions

Petar-Saric committed:
dahmmmm

Petar-Saric committed:
windows build

Petar-Saric committed:
requested changes

Petar-Saric committed:
requested changes 2

Petar-Saric committed:
testing subscribe over proxy(C++: pubnub_sample.cpp)

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
pubnub_cancel() set on every 6 minutes while testing

Petar-Saric committed:
going back to 'regular' build

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
bug fixing: using 'pb->monitor' mutex in 'pubnub_proxy.c'

Petar-Saric committed:
another correction in 'pubnub_proxy.c'

Petar-Saric committed:
another 'repeated subscribe over proxy' test(C++/samples/pubnub_sample.cpp) on travis( MacOS )

Petar-Saric committed:
nema tinyproxy koristim IP adresu javnog

Petar-Saric committed:
regular build with bug fixing in '/lib/sockets/pbpal_resolv_and_connect_sockets.c'

Petar-Saric committed:
commit bez mnogo smisla da bih mogao da spojim upstream/develop

Petar-Saric committed:
resolving conflicts in 'lib/sockets/pbpal_resolv_and_connect_sockets.c'

Petar-Saric committed:
testing repeating subscribe over some 'deaf' proxy - 'pubnub_sample.cpp' C++ openssl callback(on MacOS)

Petar-Saric committed:
catching exact 'POLLERR' moment of appearance

Petar-Saric committed:
checking if there is any 'error_code' detected by 'getsockopt()' when 'POLLHUP' shows up

Petar-Saric committed:
'error_code' from 'getsockopt()' when socket reports 'connected'(passing from 'PBS_WAIT_CONNECT' to 'PBS_CONNECTED')

Petar-Saric committed:
let's see if it works(changes in 'pbpal_check_connect()' function in 'pbpal_resolv_and_connect_sockets.c')

Petar-Saric committed:
handling 'POLLERR', 'POLLHUP' and 'POLLNVAL' when waiting for TCP socket to connect

Petar-Saric committed:
windows build

Petar-Saric committed:
updating and rearranging 'object_api' and 'signal'

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
wait_connect_timeout

Petar-Saric committed:
windows build

Petar-Saric committed:
introducing 'wait DNS response' timer the same as the one for 'wait_connect_TCP_socket'

Petar-Saric committed:
requested changes

Petar-Saric committed:
resolving 'Codacy' issue

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
pubnub objects_api

Petar-Saric committed:
requested changes

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
introduction of APPEND_URL_PARAM_TRIBOOL_SIMBOL_M() macro in 'pubnub_ccore_pubsub.h' concerning 'pubnub_history_ex()' transaction. + Preliminary Actions API

Petar-Saric committed:
multiple retries when sending DNS query, fixing await() in futreses on C++ and core/callback_samples

Petar-Saric committed:
requested changes

Petar-Saric committed:
requested changes 2

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
resolving issue: Штоперица стане после 24 дана #27

Petar-Saric committed:
let's try again

Petar-Saric committed:
requested changes + adjustments for history(), history_ex() and actions API

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
actions API adjustments

Petar-Saric committed:
requested adjustments - actions API

Petar-Saric committed:
трећа мајка сваком мора

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
requested changes

Petar-Saric committed:
resolving 'Codacy' issue

Petar-Saric committed:
requested changes 2

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

vveljko committed:
Introduce Message Actions API (#94)

- Introduce Message Actions API
- Fix a bug wherein the stopwatch module would stop working after ~25 days.
- Introduce `include_meta` options to pubnub_history_ex()

sveljko committed:
Merge remote-tracking branch 'upstream/master'

sveljko committed:
Fix Qt (#95)

Fix Qt build

sveljko committed:
Merge remote-tracking branch 'upstream/master'

sveljko committed:
Fix typo (#96)

sveljko committed:
Merge remote-tracking branch 'upstream/master'

Petar-Saric committed:
resolving conflicts

Petar-Saric committed:
resolving issue: Тупко #29 - callback_posix

Petar-Saric committed:
dahmmm

Petar-Saric committed:
игранка без престанка

Petar-Saric committed:
windows build

Petar-Saric committed:
requested changes

Petar-Saric committed:
requested changes 2

Petar-Saric committed:
requested changes 3

Petar-Saric committed:
requested changes 4

Petar-Saric committed:
updating some comments

Petar-Saric committed:
Merge remote-tracking branch 'upstream/develop' into develop

Petar-Saric committed:
Тупко #29: Windows callback - first step

Petar-Saric committed:
mmmmm

Petar-Saric committed:
Тупко #29: Windows callback - second step

Petar-Saric committed:
Тупко #29: Windows callback - third step

Petar-Saric committed:
Тупко #29: Windows callback - fourth step

sveljko commented on core/pbauto_heartbeat_callback.c:
```diff
@@ -733,48 +635,55 @@ void pbauto_heartbeat_start_timer(pubnub_t const* pb)
+        nanosleep(&sleep_time, NULL);
+    }
+#else
+    Sleep(1);
```
Ово користимо већ на неколико места. Направи помоћну `пб_слееп_мс()`, са другачијим извођењем за ПОСИЏ и Њиндоњс (у два различита модула) па је овде позови.

sveljko commented on core/pbauto_heartbeat_callback.c:
```diff
@@ -644,51 +525,72 @@ void pbauto_heartbeat_free_channelInfo(pubnub_t* pb)
+{
+    size_t len = pb_strnlen_s(str, PUBNUB_MAX_OBJECT_LENGTH);
+    char*  rslt;
+    if (len > max_size - 1) {
```
нешто не бих рекао да је ово исправна "семантика" за стрндуп(). Наиме, ако позовем са `стрндуп("Миле", 2)` очекујем да добијем показивач на `"Ми"`. Ово ће да ми врати `НУЛЛ`.

vveljko committed:
Auto heartbeat posix (#97)

User can now enable sending of heartbeat messages "in the background"
on the given context. These heartbeat messages will actually be sent
via a different context, so, won't prevent the user from sending using
her context anyway she sees fit. Heartbeat is sent according to the
latest subscribe() - if the user does a new subscribe() with new
channel/group, the heartbeats will be updated and sent accordingly.

sveljko committed:
Merge remote-tracking branch 'upstream/master'

Petar-Saric committed:
requested changes

Petar-Saric committed:
resolving conflicts
